src/arrays.js

```diff
@@ -1,44 +1,97 @@
 // Complete the following functions.
 // These functions only need to work with arrays.
 
+// each() implemented as a tail-recursive loop;
+// reduce() cannot use each() because each() does not have a default index
+const forEach = (elements, cb, i = 0) => {
```

Comment:
This is really interesting. I'm curious where you got the idea to implement `forEach` in this fashion.